pkg/tsdb/testdatasource/scenarios.go

```diff
@@ -520,59 +534,62 @@ func predictableSeries(timeRange *tsdb.TimeRange, timeStep, length int64, getVal
 	return &points, nil
 }
 
-func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResult {
+func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery, index int) *tsdb.TimeSeries {
 	timeWalkerMs := tsdbQuery.TimeRange.GetFromAsMsEpoch()
 	to := tsdbQuery.TimeRange.GetToAsMsEpoch()
+	series := newSeriesForQuery(query, index)
 
-	series := newSeriesForQuery(query)
+	startValue := query.Model.Get("startValue").MustFloat64(rand.Float64() * 100)
+	spread := query.Model.Get("spread").MustFloat64(1)
+	noise := query.Model.Get("noise").MustFloat64(0)
+	min, hasMin := query.Model.Get("min").Float64()
+	max, hasMax := query.Model.Get("max").Float64()
```

Comment:
I would change this to:
```
min, err := query.Model.Get("min").Float64()
hasMin := err == nil
max, err := query.Model.Get("max").Float64()
hasMax := err == nil
```
so the conditionals below would become:
`if hasMin && nextValue < min` and
`if hasMax && nextValue > max`